base/v0_5_exp/translate_test.go

```diff
@@ -1699,6 +1699,293 @@ func TestTranslateKernelArguments(t *testing.T) {
 	}
 }
 
+// TestTranslateSSHAuthorizedKeyResource tests translating the butane passwd.users[i].ssh_authorized_keys_local[j] entries to ignition passwd.users[i].ssh_authorized_keys[j] entries.
+func TestTranslateSSHAuthorizedKeyResource(t *testing.T) {
+	sshKeyDir := t.TempDir()
+	randomDir := t.TempDir()
+	var sshKey = "ssh-rsa AAAAAAAAA"
+	var sshKeyFileName = "id_rsa.pub"
+	var sshKeyMultipleKeysFileName = "multiple.pubs"
+	var sshKeyNonExistingFileName = "id_ed25519.pub"
+	var sshKeyEmptyFileName = "empty.pub"
+	var sshKeyBlankFileName = "blank.pub"
+	err := ioutil.WriteFile(filepath.Join(sshKeyDir, sshKeyFileName), []byte(sshKey), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+	err = ioutil.WriteFile(filepath.Join(sshKeyDir, sshKeyMultipleKeysFileName), []byte(fmt.Sprintf("%s\n%s\n\n", sshKey, sshKey)), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+	err = ioutil.WriteFile(filepath.Join(sshKeyDir, sshKeyEmptyFileName), []byte(""), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+	err = ioutil.WriteFile(filepath.Join(sshKeyDir, sshKeyBlankFileName), []byte("\n\t\n\n\n\n\n\n"), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+	tests := []struct {
+		name     string
+		in       PasswdUser
+		out      types.PasswdUser
+		report   string
+		fileDir  string
+		filePath string
+	}{
+		{
+			"empty user",
+			PasswdUser{},
+			types.PasswdUser{},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid ssh_keys_inline",
+			PasswdUser{SSHAuthorizedKeys: []SSHAuthorizedKey{SSHAuthorizedKey(sshKey)}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(sshKey)}},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid ssh_keys_local",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyFileName}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(sshKey)}},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid ssh_keys_local with multiple keys per file",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyMultipleKeysFileName}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(sshKey), types.SSHAuthorizedKey(sshKey)}},
+			"",
+			sshKeyDir,
+			"",
+		},
+		{
+			"valid ssh_keys_local and ssh_keys",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyFileName}, SSHAuthorizedKeys: []SSHAuthorizedKey{SSHAuthorizedKey(sshKey)}},
+			types.PasswdUser{SSHAuthorizedKeys: []types.SSHAuthorizedKey{types.SSHAuthorizedKey(sshKey), types.SSHAuthorizedKey(sshKey)}},
```

Comment:
These tests could be made a little more robust by making each input key different.